### PR TITLE
DOCSP-30634 Mapping Rule Options Drawer

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## Description

Mapping Rule Options drawer conversion from markdown to snooty.

## Staging

https://docs-mongodbcom-staging.corp.mongodb.com/docs-relational-migrator/docsworker-xlarge/DOCSP-30634/mapping-rules/mapping-rule-options/mapping-rule-options/

https://docs-mongodbcom-staging.corp.mongodb.com/docs-relational-migrator/docsworker-xlarge/DOCSP-30634/mapping-rules/mapping-rule-options/new-documents/

https://docs-mongodbcom-staging.corp.mongodb.com/docs-relational-migrator/docsworker-xlarge/DOCSP-30634/mapping-rules/mapping-rule-options/embedded-array/

https://docs-mongodbcom-staging.corp.mongodb.com/docs-relational-migrator/docsworker-xlarge/DOCSP-30634/mapping-rules/mapping-rule-options/embedded-documents/

## Jira

https://jira.mongodb.org/browse/DOCSP-30634

## Build

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=648c8316259a8493e5f8f1fa

### CODE REVIEW

**Comment by ianf-mongodb** [comment]
Hi @sarah-olson-mongodb this is ready for review!

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt [hunk 1]
```diff
@@ -1,8 +1,81 @@
+.. _rm-mapping-rule-options:
+
 ====================
 Mapping Rule Options
 ====================
 
+Mapping rule options are important options that help define the state
+of your MongoDB collection after a Migration.
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/mapping-rule-options.txt:8]:
```suggestion
of your MongoDB collection after a migration.
```

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt [hunk 2]
```diff
@@ -1,8 +1,81 @@
...
+     - Description
+     - Applies To
+   * - Collection Name
+     - The destination collection where the relational rows of data are migrated.
+     - :guilabel:`New documents` 
+   * - Parent collection
+     - The destination collection where the relational rows of data are migrated.
+     - :guilabel:`Embedded documents` :guilabel:`Embedded array` 
+   * - Prefix
+     - A drop-down where you can choose where in the hierarchy the rule goes.
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/mapping-rule-options.txt:43]: (wording)
Consider something simpler. Maybe:
Specify your rule hierarchy.

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt [hunk 3]
```diff
@@ -1,8 +1,81 @@
...
+     - :guilabel:`Embedded documents` :guilabel:`Embedded array` 
+   * - Field name
+     - A text field which indicates the field that contains the array or document.
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/mapping-rule-options.txt:46]: (wording)
Consider something like: Indicates the field containing the array or document.

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt [hunk 4]
```diff
@@ -1,8 +1,81 @@
...
+     - :guilabel:`Embedded documents` :guilabel:`Embedded array` 
+   * - Create an array of primitive values
+     - Creates an array of a single field in the parent collection.
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/mapping-rule-options.txt:49]: (wording)
Creates a single field array in the parent collection.

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt [hunk 5]
```diff
@@ -1,8 +1,81 @@
...
+     - :guilabel:`Embedded array` 
+   * - Merge fields into the parent
+     - Merge the table rows as fields at the parent collection level.
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/mapping-rule-options.txt:52]:
```suggestion
     - Merges the table rows as fields at the parent collection level.
```

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt [hunk 6]
```diff
@@ -1,8 +1,81 @@
...
+     - :guilabel:`Embedded documents` 
+
+.. note::
+
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/mapping-rule-options.txt:56]: (wording)
Just a minor tweak. Consider something like:
Relational Migratory primary key id field migration is defined at the project level.

File: source/mapping-rules/mapping-rule-options/new-documents.txt [hunk 7]
```diff
@@ -2,4 +2,83 @@
...
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+
+The :guilabel:`New Documents` mapping rule is the most basic mapping rule option
+which creates a document for each row of data in your relational database.
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/new-documents.txt:15]: (simple sentences)
Consider something like:
"New Documents is the most basic mapping rule option. It creates a document for each row of data in your relational database.

File: source/mapping-rules/mapping-rule-options/new-documents.txt [hunk 8]
```diff
@@ -2,4 +2,83 @@
...
+Any database table can use a :guilabel:`New Documents` mapping rule. 
+
+Behavior
+--------
+
+Use this mapping rule when you want data from a table to be 
+stored in a dedicated collection. 
+
+You can have multiple :guilabel:`New Documents` mapping rules from
+different tables into the same collection. Relational Migrator will treat
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/new-documents.txt:29]: (present tense)
```suggestion
different tables into the same collection. Relational Migrator treats
```

File: source/mapping-rules/mapping-rule-options/new-documents.txt [hunk 9]
```diff
@@ -2,4 +2,83 @@
...
+these tables as isolated data and the destination documents will not be
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/new-documents.txt:30]:
```suggestion
these tables as isolated data and the destination documents are not 
```

File: source/mapping-rules/mapping-rule-options/new-documents.txt [hunk 10]
```diff
@@ -2,4 +2,83 @@
...
+
+New Documents
+~~~~~~~~~~~~~
+
+In this example a :guilabel:`New Documents` mapping rule is applied to a single row 
+from the ``Customer`` table.
+
+
+Click the tabs below to see the input and output for this example:
+
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/new-documents.txt:47]: (praise) 
I like this use of tabs a lot!

File: source/mapping-rules/mapping-rule-options/embedded-array.txt [hunk 11]
```diff
@@ -1,5 +1,122 @@
...
 ==============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+The :guilabel:`Embedded array` mapping rule is an advanced mapping 
+rule option which must be explicitly added to a mapping model.
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/embedded-array.txt:14]: (nit)
Consider "... is an advanced mapping rule option. It must be ... "

File: source/mapping-rules/mapping-rule-options/embedded-array.txt [hunk 12]
```diff
@@ -1,5 +1,122 @@
...
+
+About this Task
+---------------
+
+The :guilabel:`Embedded array` mapping rule is available if the relational
+table it is mapping from is on the many side of the foreign key relationship. 
+
+The table which contains the primary key must also be mapped to a
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/embedded-array.txt:22]:
```suggestion
The table containing the primary key must also be mapped to a 
```

File: source/mapping-rules/mapping-rule-options/embedded-array.txt [hunk 13]
```diff
@@ -1,5 +1,122 @@
...
+           - OrderStatusID
+         * - 1
+           - 1
+           - 50.00
+           - 1
+         * - 4
+           - 1
+           - 500.00
+           - 3
+
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/embedded-array.txt:89]: (praise)
Really nice example here!

File: source/mapping-rules/mapping-rule-options/embedded-documents.txt [hunk 14]
```diff
@@ -1,3 +1,257 @@
...
+==================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+The :guilabel:`Embedded Documents` mapping rule is an advanced mapping 
+rule option which must be explicitly added to a mapping model.
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/embedded-documents.txt:14]: (nit)
Consider "... mapping rule option. It must be ..."

File: source/mapping-rules/mapping-rule-options/embedded-documents.txt [hunk 15]
```diff
@@ -1,3 +1,257 @@
...
+
+About this Task
+---------------
+
+The :guilabel:`Embedded Documents` mapping rule is available if the relational
+table it is mapping from is on the one side of the foreign key relationship.
+
+The table which contains the foreign key must also be mapped to a
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/embedded-documents.txt:22]: (nit)
The table containing the ...

File: source/mapping-rules/mapping-rule-options/embedded-documents.txt [hunk 16]
```diff
@@ -1,3 +1,257 @@
...
+--------
+
+This section demonstrates the relational input and
+the MongoDB output of the :guilabel:`Embedded Documents` mapping rule.
+
+Embedded Documents
+~~~~~~~~~~~~~~~~~~
+
+In this example:
+
```
Comment (sarah-olson-mongodb) [inline source/mapping-rules/mapping-rule-options/embedded-documents.txt:51]: (praise)
This is a really nice use of a list to describe the example behavior. It helps, IMO, to have that setup in advance of the toggle.

**Review by sarah-olson-mongodb — COMMENTED** [review]
Thanks a ton @ianf-mongodb! I made a few comments and suggestions. Would you mind sending it back to me for an additional pass tomorrow. I don't think we refactored this one and I'd love to take one more super quick look!

File: source/mapping-rules/mapping-rule-options/embedded-array.txt (hunk 12 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/embedded-array.txt]: Updated.

File: source/mapping-rules/mapping-rule-options/embedded-documents.txt (hunk 15 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/embedded-documents.txt]: Updated.

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt (hunk 2 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/mapping-rule-options.txt]: Good suggestion, thanks.

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt (hunk 3 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/mapping-rule-options.txt]: Updated.

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt (hunk 4 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/mapping-rule-options.txt]: Updated.

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt (hunk 5 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/mapping-rule-options.txt]: Updated.

File: source/mapping-rules/mapping-rule-options/mapping-rule-options.txt (hunk 6 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/mapping-rule-options.txt:56]: Updated.

File: source/mapping-rules/mapping-rule-options/new-documents.txt (hunk 7 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/new-documents.txt]: Yes, good suggestion. Updated.

File: source/mapping-rules/mapping-rule-options/new-documents.txt (hunk 8 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/new-documents.txt]: Updated.

File: source/mapping-rules/mapping-rule-options/new-documents.txt (hunk 9 above)
Comment (ianf-mongodb) [reply source/mapping-rules/mapping-rule-options/new-documents.txt]: Updated.

**Review by sarah-olson-mongodb — APPROVED** [review]
Thanks @ianf-mongodb! This looks great!
LGTM

**Pull request merged**